samples/subsys/usb/cdc_acm/src/main.c

```diff
@@ -41,12 +41,27 @@ static void interrupt_handler(struct device *dev)
 
 static void write_data(struct device *dev, const char *buf, int len)
 {
+	int sent;
+	u32_t transfer_len;
 	uart_irq_tx_enable(dev);
 
-	data_transmitted = false;
-	uart_fifo_fill(dev, (const u8_t *)buf, len);
-	while (data_transmitted == false)
-		;
+	while (len) {
+		data_transmitted = false;
+		/* FIXME: An issue observed when USB TX Endpoint is written
+		 * more than once (greater than 4 Bytes of data is written)
+		 * (Jira 2074). So restricting writing 4 bytes at a time into
+		 * USB TX Endpoing.
+		 */
+		transfer_len = len > sizeof(u32_t) ? sizeof(u32_t) : len;
+		sent = uart_fifo_fill(dev, (const u8_t *)buf, transfer_len);
```

Comment:
Here is new versions of cdc_acm_fifo_fill if we can accept hack in subsystem. 
But as per my understanding we should leave it to application to handle it instead of putting hack in driver/subsystem.

static int cdc_acm_fifo_fill(struct device *dev,
                             const u8_t *tx_data, int len)
{
        struct cdc_acm_dev_data_t * const dev_data = DEV_DATA(dev);
        u32_t bytes_written = 0, total_bytes_written = 0, transfer_len = 0;


        if (dev_data->usb_status != USB_DC_CONFIGURED) {
                return 0;
        }

        while(len) {
                dev_data->tx_ready = 0;
                transfer_len = len > sizeof(u32_t) ? sizeof(u32_t) : len;
                usb_write(CDC_ENDP_IN, tx_data, transfer_len, &bytes_written);

                while(!dev_data->tx_ready)
                        ;

                len -= bytes_written;
                tx_data += bytes_written;
                total_bytes_written += bytes_written;
        }
        return total_bytes_written;
}

For this we need to make another change inside cdc_acm_irq_tx_ready (snap shot of code given below), we need to avoid modifying "dev_data->tx_ready", 
As per my understanding it should only return the current status of dev_data->tx_ready. 
static int cdc_acm_irq_tx_ready(struct device *dev)
{
        struct cdc_acm_dev_data_t * const dev_data = DEV_DATA(dev);
        if (dev_data->tx_ready) {
               //dev_data->tx_ready = 0;
                return 1;
        }
        return 0;
}